plugin/trino-hive/src/main/java/io/trino/plugin/hive/projection/PartitionProjection.java

```diff
@@ -44,14 +46,58 @@ public final class PartitionProjection
     private static final Pattern PROJECTION_LOCATION_TEMPLATE_PLACEHOLDER_PATTERN = Pattern.compile("(\\$\\{[^}]+\\})");
 
     private final Optional<String> storageLocationTemplate;
+    private final boolean templateCompatibleWithHive;
```

Comment:
Nit: maybe name this field after the specific action taken, e.g.: `escapeInjectedColumnValues`